Sources/JWTKit/EdDSA/EdDSAKey.swift

```diff
@@ -0,0 +1,51 @@
+import Foundation
+import Crypto
+
+public struct EdDSAKey {
+            
+    let keyPair: OctetKeyPair
+    var publicKey: Data? {
+        keyPair.publicKey
+    }
+    var privateKey: Data? {
+        keyPair.privateKey
+    }
+    let curve: JWK.Curve
+    
+    public init(x: String?, d: String? = nil, curve: JWK.Curve = .ed25519) throws {
```

Comment:
You can technically construct this object with a public key and a private key that don't match. That seems bad.

Reply:
Is there a way to guarantee/verify that the keypair matches when provided like in https://www.rfc-editor.org/rfc/rfc8037.html#appendix-A

Reply:
Yes: in the format you've linked to, the public key is provided as part of the private key.

Reply:
Private and public keys are distinct now. The curve is not falling back to `.ed25519` anymore.
https://github.com/vapor/jwt-kit/pull/82/commits/12f9a890c5a5829b10c1ff262c128288e745e5cb